crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -179,37 +173,48 @@ impl Plugin for Core3dPlugin {
     }
 }
 
+/// Opaque 3D [`BinnedPhaseItem`]s.
 pub struct Opaque3d {
-    pub asset_id: AssetId<Mesh>,
-    pub pipeline: CachedRenderPipelineId,
-    pub entity: Entity,
-    pub draw_function: DrawFunctionId,
+    /// The key, which determines which can be batched.
+    pub key: Opaque3dBinKey,
+    /// An entity from which the mesh and other attributes will be fetched.
```

Comment:
Strictly speaking, Opaque3d isn't coupled with meshes, so this might need to be reworded.

Reply:
Done